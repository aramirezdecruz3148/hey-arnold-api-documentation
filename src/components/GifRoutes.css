@import url('https://fonts.googleapis.com/css?family=Luckiest+Guy&display=swap');

.flex {
  display: flex;
  flex-direction: column;
  margin-left: 3vw;
}

.codeBlock {
  padding-top: 10px;
  height: 20vh;
  overflow: scroll;
  list-style: none;
  width: 75vw;
  background: whitesmoke;
}

.title {
  font-family: 'Luckiest Guy', cursive;
  color: #FFD90F;
  font-size: 3vw;
  margin-bottom: -.5vw;
  text-shadow:
		-.2vw -.2vw 0 #000,
		.2vw -.2vw 0 #000,
		-.2vw .2vw 0 #000,
    .2vw .2vw 0 #000;
}

.endpoints {
  font-size: 1.6vw;
  font-style: italic;
}

.queries {
  background: #cebfc4;
  color: #4d4e5c;
  width: 15vw;
  padding: .5vw;
  text-align: center;
  font-size: 1.7vw;
}

.description {
  font-size: 1.5vw;
}