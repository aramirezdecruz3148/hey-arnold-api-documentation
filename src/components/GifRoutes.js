import React from 'react';

export default function GifRoutes() {
  return (
    <>
      <h4>Gif Routes</h4>
      <p>
      This is where I will talk about the Gif
      route endpoints, have a couple code block examples
      of the res.body, and maybe have a table with the endpoints.
      </p>
    </>
  );
}
