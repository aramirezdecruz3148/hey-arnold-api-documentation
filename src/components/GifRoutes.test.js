import React from 'react';
import { shallow } from 'enzyme';
import GifRoutes from './GifRoutes';

describe('GifRoutes component', () => {
  it('renders GifRoutes', () => {
    const wrapper = shallow(<GifRoutes />);
    expect(wrapper).toMatchSnapshot();
  });
});
