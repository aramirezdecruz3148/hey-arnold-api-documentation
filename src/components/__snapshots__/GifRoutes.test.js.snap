// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifRoutes component renders GifRoutes 1`] = `
<div
  className="flex"
>
  <h3
    className="title"
  >
    Gif Routes
  </h3>
  <section>
    <h4
      className="queries"
    >
      Get All GIFs
    </h4>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/gifs
    </h5>
    <p
      className="description"
    >
      This route will return a list of 49 Hey Arnold GIFs.
    </p>
  </section>
  <section>
    <h4
      className="queries"
    >
      Get Random GIF
    </h4>
    <p
      className="description"
    >
      This route returns one random GIF by default. To change this number, simply add a 
      <q>
        count
      </q>
       query parameter, e.g. ?count=2.
    </p>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/gifs/random?count=2
    </h5>
    <div
      className="codeBlock"
    >
      <pre>
        [
  {
    "_id": "5da23754845fd2cb76d59f10",
    "gifLink": "https://media.tenor.com/images/7511947a6ae9acb62c9515bfba14b1ec/tenor.gif"
  },
  {
    "_id": "5da23754845fd2cb76d59f05",
    "gifLink": "https://media.tenor.com/images/369dabafda46a71e9fdea7ff6d89fc6d/tenor.gif"
  }
]
      </pre>
    </div>
  </section>
  <section>
    <h4
      className="queries"
    >
      Get GIF by Id
    </h4>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/gifs/5da23754845fd2cb76d59ee7
    </h5>
    <div
      className="codeBlock"
    >
      <pre>
        {
  "_id": "5da23754845fd2cb76d59ee7",
  "gifLink": "https://media.tenor.com/images/4393cee845d8464670817e40b0601e3b/tenor.gif"
}
      </pre>
    </div>
  </section>
</div>
`;
